sdk/core/azure-core/src/main/java/com/azure/core/http/rest/ResponseConstructorsCache.java

```diff
@@ -67,8 +64,12 @@ private MethodHandle locateResponseConstructor(Class<?> responseClass) {
         MethodHandles.Lookup lookupToUse;
         try {
             lookupToUse = ReflectionUtilsApi.INSTANCE.getLookupToUse(responseClass);
-        } catch (Throwable t) {
-            throw LOGGER.logExceptionAsError(new RuntimeException(t));
+        } catch (Exception ex) {
+            if (ex instanceof RuntimeException) {
+                throw LOGGER.logExceptionAsError((RuntimeException) ex);
+            }
+
+            throw LOGGER.logExceptionAsError(new RuntimeException(ex));
```

Comment:
>logExceptionAsError [](http://example.com/codeflow?start=25&length=19)

If we should never catch a throwable should we even expose logThrowableAsError() as a public API?

Reply:
Turns out we named these APIs poorly, `logExceptionAs*` is for `RuntimeException`s and should have been `logRuntimeExceptionAs*`, `logThrowableAs*` supports checked `Exception`s